analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDatabasePlugin.java

```diff
@@ -0,0 +1,244 @@
+                                null, null, null);
+                    }
+                    depIds.add(depId);
+                }
+            }
+            if (depIds.size() > 0) {
+                metadataDao.insertDependencies(packageVersionId, depIds, depVersions);
+            }
+
+            var cha = callGraph.getClassHierarchy();
+            var globalIdsMap = new HashMap<Integer, Long>();
+            for (var fastenUri : cha.keySet()) {
+                var type = cha.get(fastenUri);
+                var fileMetadata = new JSONObject();
+                fileMetadata.put("superInterfaces",
+                        ExtendedRevisionCallGraph.Type.toListOfString(type.getSuperInterfaces()));
+                fileMetadata.put("sourceFile", type.getSourceFileName());
+                fileMetadata.put("superClasses",
+                        ExtendedRevisionCallGraph.Type.toListOfString(type.getSuperClasses()));
+                long fileId = metadataDao.insertFile(packageVersionId, fastenUri.getNamespace(),
+                        null, null, fileMetadata);
+                for (var methodEntry : type.getMethods().entrySet()) {
+                    var uri = methodEntry.getValue().toString();
+                    long callableId = metadataDao.insertCallable(fileId, uri, true, null, null);
+                    globalIdsMap.put(methodEntry.getKey(), callableId);
+                }
+            }
+
+            var graph = callGraph.getGraph();
+
+            var internalCalls = graph.getInternalCalls();
+            for (var call : internalCalls) {
+                var sourceLocalId = call.get(0);
+                var targetLocalId = call.get(1);
+                var sourceGlobalId = globalIdsMap.get(sourceLocalId);
+                var targetGlobalId = globalIdsMap.get(targetLocalId);
+                metadataDao.insertEdge(sourceGlobalId, targetGlobalId, null);
+            }
+
+            var externalCalls = graph.getExternalCalls();
+            for (var callEntry : externalCalls.entrySet()) {
+                var call = callEntry.getKey();
+                var sourceLocalId = call.getKey();
+                var sourceGlobalId = globalIdsMap.get(sourceLocalId);
+                var uri = call.getValue().toString();
+                var targetId = metadataDao.insertCallable(null, uri, false, null, null);
+                var edgeMetadata = new JSONObject(callEntry.getValue());
+                metadataDao.insertEdge(sourceGlobalId, targetId, edgeMetadata);
+            }
+        }
+
+        @Override
+        public boolean recordProcessSuccessful() {
+            return this.processedRecord;
+        }
+
+        @Override
+        public String name() {
+            return "Metadata plugin";
+        }
+
+        @Override
+        public String description() {
+            return "Metadata plugin. "
+                    + "Consumes kafka topic and populates metadata database with consumed data.";
```

Comment:
"Consumes _ExtendedRevisionCallgraph_-formatted JSON objects from a Kafka topic..._

Reply:
Changed it